src/Sarif.Viewer.VisualStudio/Telemetry/Events.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Sarif.Viewer.Telemetry
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Contains the methods for firing telemetry events from the SARIF viewer extension.
+    /// </summary>
+    /// <remarks>
+    /// The documentation in this class is super useful for understanding where this telemetry is fired.
+    /// The documentation should also contain "why" the telemetry is being fired, i.e., why is this data needed?
```

Comment:
You mention "the why", but I don't see it being adhered to below.

Reply:
Now I understand, it's because you're using `[CallerMemberName]`.

---
In reply to: [505749749](https://github.com/microsoft/sarif-visualstudio-extension/pull/226#discussion_r505749749) [](ancestors = 505749749)

Reply:
You are correct. And I will expand the comment and probably remove ALL telemetry points.

Right now I would say that there are no "whys" for any of these.

We shouldn't be firing telemetry for data we will never consume.

So, we have the option to check in with NO telemetry points until we understand "why" we want this data.

For example, it isn't interesting to know that someone called an API, or that something was executed unless you plan to take action based on that data. Windows (for example) takes a hard stance on this since telemetry takes not only user bandwidth but storage space on Azure.

---
In reply to: [505749749](https://github.com/microsoft/sarif-visualstudio-extension/pull/226#discussion_r505749749) [](ancestors = 505749749)